src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/ConditionalWeakTable.cs

```diff
@@ -157,7 +162,38 @@ public bool Remove(TKey key)
 
             lock (_lock)
             {
-                return _container.Remove(key);
+                return _container.Remove(key, out _);
+            }
+        }
+
+        /// <summary>Removes a key and its value from the table, and returns the removed value if it was present.</summary>
+        /// <param name="key">key to remove. May not be null.</param>
+        /// <param name="value">value removed from the table, if it was present.</param>
+        /// <returns>true if the key is found and removed. Returns false if the key was not in the dictionary.</returns>
+        /// <exception cref="ArgumentNullException"><paramref name="key"/> is <see langword="null"/>.</exception>
```

Comment:
```suggestion
        /// <summary>Removes a key and its value from the table, and returns the removed value if it was present.</summary>
        /// <param name="key">The key to remove.</param>
        /// <param name="value">value removed from the table, if it was present.</param>
        /// <returns><see langword="true"/> if the key is found and removed; otherwise, <see langword="false"/>.</returns>
        /// <exception cref="ArgumentNullException"><paramref name="key"/> is <see langword="null"/>.</exception>
```